### PR TITLE
Declare TypeScript language

### LINKED ISSUE #1288: Declare TypeScript language

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #1288

### CODE REVIEW

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java [hunk 1]
```diff
@@ -74,6 +73,9 @@
 
   public static final String DEPRECATED_ESLINT_PROPERTY = "sonar.typescript.eslint.reportPaths";
 
+  private static final String TYPESCRIPT_CATEGORY = "TypeScript";
+  private static final String GENERAL_SUBCATEGORY = "General";
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: we already have a constant for this one

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java [hunk 2]
```diff
@@ -177,5 +179,22 @@ public void define(Context context) {
...
 
+    context.addExtensions(typeScriptExtensions());
+  }
+
+  private static Collection<?> typeScriptExtensions() {
+    return Arrays.asList(
+      TypeScriptLanguage.class,
+
+      PropertyDefinition.builder(TypeScriptLanguage.FILE_SUFFIXES_KEY)
+      .defaultValue(TypeScriptLanguage.FILE_SUFFIXES_DEFVALUE)
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: why there is no indentation here?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptLanguage.java [hunk 3]
```diff
@@ -1,46 +1,48 @@
...
+
+  private Configuration configuration;
+
+  public JavaScriptLanguage(Configuration configuration) {
+    super(KEY, "JavaScript");
+    this.configuration = configuration;
+  }
+
+  @Override
+  public String[] getFileSuffixes() {
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptLanguage.java]: should we simplify this implementation the same way it's done for TS?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java [hunk 4]
```diff
@@ -99,8 +101,8 @@ public void define(Context context) {
         .multiValues(true)
         .build(),
 
-      PropertyDefinition.builder(FILE_SUFFIXES_KEY)
-        .defaultValue(FILE_SUFFIXES_DEFVALUE)
+      PropertyDefinition.builder(JavaScriptLanguage.FILE_SUFFIXES_KEY)
+        .defaultValue(JavaScriptLanguage.FILE_SUFFIXES_DEFVALUE)
         .name("File Suffixes")
         .description("List of suffixes for files to analyze.")
```
Comment (quentin-jaquier-sonarsource) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: We should also update this description, to match the one at line [192](https://github.com/SonarSource/SonarJS/pull/1295/files#diff-75c0e6bce6afa14a48c230e8f54b880bR192), in order to be clear with the user if he can or cannot use comma.

Comment (vilchik-elena) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: I would say the opposite: this description is ok, the one for TS should be updated as this description is only seen in UI where commas are not allowed

Comment (saberduck) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: I think I am with Lena here, I updated both descriptions to be "List of suffixes for files to analyze."

**Pull request merged**